vllm/sequence.py

```diff
@@ -1259,6 +1262,12 @@ def expand_with_bonus_tokens(
             [self.hidden_states, self.second_last_token_hidden_states])[index]
 
 
+@dataclass
+class AsyncCallbackData:
+    func: Callable
+    kw_args: Dict[str, Any]
```

Comment:
Could use `functools.partial` for this rather than a custom class?

Reply:
Oh good idea, changed to use partial and the code is simpler now, thanks for the suggestion!